### PR TITLE
docs: clarify the new default client cipher list

### LINKED ISSUE #16469: doc: default cipher suite docs are misleading for server and client context
*Title*: default cipher suite docs are misleading for server and client context

*Description*:

 The [default tls client config](https://github.com/envoyproxy/envoy/pull/13850) has changed since v1.16 to be distinct from the default server config. 

From

https://github.com/envoyproxy/envoy/blob/bf5d0eb44b781ac26ff1513700bcb114b7cf4300/source/extensions/transport_sockets/tls/context_config_impl.cc#L332-L356

To

https://github.com/envoyproxy/envoy/blob/ce071f6a7f5524e09c5d7bac75c9951e1b827a50/source/extensions/transport_sockets/tls/context_config_impl.cc#L315-L331

This is not clear in the [docs](https://www.envoyproxy.io/docs/envoy/v1.17.3/api-v3/extensions/transport_sockets/tls/v3/common.proto.html?highlight=tls#extensions-transport-sockets-tls-v3-tlsparameters). 

For example, where `tls_minimum_protocol_version` / `tls_maximum_protocol_version` explicitly mention the difference in client and server configs, `cipher_suites` does not.

If you agree, I'd be happy to make this change / update the docs to be more clear that the new default client cipher suite is as follows. 

```
[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]
[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-RSA-AES256-GCM-SHA384
```

### Related context

- https://github.com/pomerium/pomerium/issues/2195
- https://github.com/envoyproxy/envoy/issues/5401 
- #13850
- #5396
- #5397

### PR DESCRIPTION
**Pull request by desimone**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/main/PULL_REQUESTS.md)

Commit Message: docs: clarify the new default client cipher list
Additional Description: These changes clarify that as of v1.16 the default cipher suite is different for client and servers.
Risk Level: Low
Testing: N/A
Docs Changes: Yes
Release Notes: N/A
Platform Specific Features: N/A 

Fixes #16469

### CODE REVIEW

**Review by PiotrSikora — DISMISSED** [review]
You need to fix format (run `./ci/run_envoy_docker.sh './ci/do_ci.sh fix_format'`, `./ci/do_ci.sh fix_format` or `./tools/proto_format/proto_format.sh fix`), otherwise LGTM. Thanks!

**Comment by antoniovicente** [comment]
cc @htuch 

These changes need to be applied to https://github.com/envoyproxy/envoy/blob/d26b16c328e594e80ea7fba6243e336d2d2be6aa/api/envoy/extensions/transport_sockets/tls/v3/common.proto

And the scripts that Piotr mentions in https://github.com/envoyproxy/envoy/pull/16474#pullrequestreview-659652331 need to be run to regenerate shadows from the canonical source mentioned above.

**Comment by desimone** [comment]
Reapplied the changes to the correct proto. 

\cc @travisgroth can you help me with an assist on those scripts? They don't work with my M1 processor :/

File: api/envoy/extensions/transport_sockets/tls/v3/common.proto
Comment (htuch) [inline api/envoy/extensions/transport_sockets/tls/v3/common.proto:109]: It would be nice if we could generate these from some canonical location that is shared with code. If we aren't going to change frequently this is fine though.

**Review by htuch — COMMENTED** [review]
Thanks, defer to @PiotrSikora to sign-off on the correctness of the cipher lists.

**Comment by PiotrSikora** [comment]
> Thanks, defer to @PiotrSikora to sign-off on the correctness of the cipher lists.

I already did (https://github.com/envoyproxy/envoy/pull/16474#pullrequestreview-659652331).

File: api/envoy/extensions/transport_sockets/tls/v3/common.proto
Comment (htuch) [inline api/envoy/extensions/transport_sockets/tls/v3/common.proto:59]: Nit: s/servers/server/

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo nit. Please fix format as suggested to @PiotrSikora

**Comment by travisgroth** [comment]
@PiotrSikora I'm getting an error running the format commands:

```
% ./tools/proto_format/proto_format.sh fix
api/envoy/extensions/transport_sockets/tls/v3/common.proto
Starting local Bazel server and connecting to it...
INFO: SHA256 (https://golang.org/dl/?mode=json&include=all) = c71d9d3a5e2d4baaa5ae93319092dde1d43670c2ddf0ee38507695792febe51d
INFO: Analyzed target @envoy_api_canonical//versioning:active_protos (337 packages loaded, 2034 targets configured).
INFO: Found 1 target...
INFO: From CcCmakeMakeRule external/envoy/bazel/foreign_cc/zlib/include [for host]:

INFO: From CcCmakeMakeRule external/envoy/bazel/foreign_cc/zlib/include:

INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/empty_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/compiler/plugin_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_googleapis/google/api/annotations_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/wrappers_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile tools/type_whisperer/types_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/com_github_cncf_udpa/udpa/annotations/status_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/struct_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/envoy_api_canonical/envoy/annotations/deprecation_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/duration_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_googleapis/google/api/httpbody_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/timestamp_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/source_context_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_github_cncf_udpa/udpa/annotations/security_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/envoy_api_canonical/envoy/annotations/resource_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/com_github_cncf_udpa/udpa/annotations/versioning_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_googleapis/google/api/http_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_github_cncf_udpa/udpa/annotations/sensitive_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile tools/type_whisperer/api_type_db_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/com_envoyproxy_protoc_gen_validate/validate/validate_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/any_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/type_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/descriptor_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_github_cncf_udpa/udpa/annotations/migrate_pb2.py:
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_envoyproxy_protoc_gen_validate: warning: directory does not exist.
external/com_google_protobuf/python: warning: directory does not exist.
bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_googleapis: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/api_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_googleapis/google/rpc/status_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/field_mask_pb2.py:
external/com_google_protobuf/python: warning: directory does not exist.
INFO: From TypeWhisperer external/com_google_protobuf/descriptor_proto/google/protobuf/descriptor.proto.types.pb_text:
./external/com_google_protobuf: warning: directory does not exist.
INFO: From TypeWhisperer external/com_google_protobuf/any_proto/google/protobuf/any.proto.types.pb_text:
./external/com_google_protobuf: warning: directory does not exist.
INFO: From TypeWhisperer external/com_google_protobuf/wrappers_proto/google/protobuf/wrappers.proto.types.pb_text:
./external/com_google_protobuf: warning: directory does not exist.
INFO: From TypeWhisperer external/com_google_protobuf/duration_proto/google/protobuf/duration.proto.types.pb_text:
./external/com_google_protobuf: warning: directory does not exist.
INFO: From TypeWhisperer external/com_google_protobuf/timestamp_proto/google/protobuf/timestamp.proto.types.pb_text:
./external/com_google_protobuf: warning: directory does not exist.
INFO: From TypeWhisperer external/com_google_protobuf/struct_proto/google/protobuf/struct.proto.types.pb_text:
./external/com_google_protobuf: warning: directory does not exist.
INFO: From TypeWhisperer external/com_google_protobuf/empty_proto/google/protobuf/empty.proto.types.pb_text:
./external/com_google_protobuf: warning: directory does not exist.
ERROR: /private/var/tmp/_bazel_tgroth/317f6559654b1ca2fa839199a5ffc255/external/com_google_protobuf/BUILD:356:15: protoxform external/com_google_protobuf/struct_proto/google/protobuf/struct.proto.active_or_frozen.proto failed (Exit 1): protoc failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/protoc -I./external/com_google_protobuf ... (remaining 6 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox protoc failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/protoc -I./external/com_google_protobuf ... (remaining 6 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
./external/com_google_protobuf: warning: directory does not exist.
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_tgroth/317f6559654b1ca2fa839199a5ffc255/sandbox/darwin-sandbox/672/execroot/envoy/bazel-out/darwin-opt-exec-2B5CBBC6/bin/tools/protoxform/protoxform.runfiles/envoy/tools/protoxform/protoxform.py", line 11, in <module>
    from tools.protoxform import migrate, utils
  File "/private/var/tmp/_bazel_tgroth/317f6559654b1ca2fa839199a5ffc255/sandbox/darwin-sandbox/672/execroot/envoy/bazel-out/darwin-opt-exec-2B5CBBC6/bin/tools/protoxform/protoxform.runfiles/envoy/tools/protoxform/migrate.py", line 14, in <module>
    from google.api import annotations_pb2
ModuleNotFoundError: No module named 'google.api'
--api_proto_plugin_out: protoc-gen-api_proto_plugin: Plugin failed with status code 1.
Aspect //tools/protoxform:protoxform.bzl%protoxform_aspect of @envoy_api_canonical//versioning:active_protos failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 139.672s, Critical Path: 37.57s
INFO: 707 processes: 41 internal, 666 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

Is there any setup needed to run that script?  I tried via docker, which also failed, but it gave me no error output to triage.

File: api/envoy/extensions/transport_sockets/tls/v3/common.proto
Comment (desimone) [reply api/envoy/extensions/transport_sockets/tls/v3/common.proto]: ✅

**Comment by PiotrSikora** [comment]
> ```
> Use --sandbox_debug to see verbose messages from the sandbox
> ./external/com_google_protobuf: warning: directory does not exist.
> Traceback (most recent call last):
>   File "/private/var/tmp/_bazel_tgroth/317f6559654b1ca2fa839199a5ffc255/sandbox/darwin-sandbox/672/execroot/envoy/bazel-out/darwin-opt-exec-2B5CBBC6/bin/tools/protoxform/protoxform.runfiles/envoy/tools/protoxform/protoxform.py", line 11, in <module>
>     from tools.protoxform import migrate, utils
>   File "/private/var/tmp/_bazel_tgroth/317f6559654b1ca2fa839199a5ffc255/sandbox/darwin-sandbox/672/execroot/envoy/bazel-out/darwin-opt-exec-2B5CBBC6/bin/tools/protoxform/protoxform.runfiles/envoy/tools/protoxform/migrate.py", line 14, in <module>
>     from google.api import annotations_pb2
> ModuleNotFoundError: No module named 'google.api'
> ```
> 
> Is there any setup needed to run that script? I tried via docker, which also failed, but it gave me no error output to triage.

It looks that you're missing `google-api-python-client`. I don't believe we have any setup script for that, unfortunately, but you can install it using `pip install google-api-python-client`.

It's weird that the Docker would fail, since it should be self-contained, and it works fine for me:
```
$ git checkout desimone/16469
```
```
$ ./ci/run_envoy_docker.sh './ci/do_ci.sh fix_format'
[...]
```
``` 
$ git diff --stat
 api/envoy/extensions/transport_sockets/tls/v4alpha/common.proto                  | 28 ++++++++++++++++++++++++----
 generated_api_shadow/envoy/extensions/transport_sockets/tls/v3/common.proto      | 28 ++++++++++++++++++++++++----
 generated_api_shadow/envoy/extensions/transport_sockets/tls/v4alpha/common.proto | 28 ++++++++++++++++++++++++----
 3 files changed, 72 insertions(+), 12 deletions(-)
```

**Comment by htuch** [comment]
Maybe @travisgroth isn't running in Docker..

**Comment by travisgroth** [comment]
@htuch 

```shell
% ./ci/run_envoy_docker.sh './ci/do_ci.sh fix_format'
No remote cache is set, skipping setup remote cache.
ENVOY_SRCDIR=/source
ENVOY_BUILD_TARGET=//source/exe:envoy-static
ENVOY_BUILD_ARCH=x86_64
$TEST_TMPDIR defined: output root default is '/build/tmp' and max_idle_secs default is '15'.
Starting local Bazel server and connecting to it...
Skip setting up Envoy Filter Example.
building using 4 CPUs
building for x86_64
clang toolchain with libc++ configured
fix_format...
api/envoy/extensions/transport_sockets/tls/v3/common.proto
protoxform_test...
$TEST_TMPDIR defined: output root default is '/build/tmp' and max_idle_secs default is '15'.
Starting local Bazel server and connecting to it...
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 2 packages loaded
Loading: 2 packages loaded
$TEST_TMPDIR defined: output root default is '/build/tmp' and max_idle_secs default is '15'.
INFO: Analyzed target //tools/testdata/protoxform:fix_protos (65 packages loaded, 1241 targets configured).
INFO: Found 1 target...
INFO: From CcCmakeMakeRule external/envoy/bazel/foreign_cc/zlib/include [for host]:

ERROR: /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/bazel/foreign_cc/BUILD:349:21: TreeArtifact external/envoy/bazel/foreign_cc/zlib/include was not created
ERROR: /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/bazel/foreign_cc/BUILD:349:21: TreeArtifact external/envoy/bazel/foreign_cc/copy_zlib/zlib was not created
ERROR: /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/bazel/foreign_cc/BUILD:349:21: not all outputs were created or valid
Aspect //tools/protoxform:protoxform.bzl%protoxform_aspect of //tools/testdata/protoxform:fix_protos failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 13.459s, Critical Path: 4.92s
INFO: 27 processes: 20 internal, 7 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

**Comment by travisgroth** [comment]
@PiotrSikora thanks that did the trick.

**Review by PiotrSikora — APPROVED** [review]
Thanks!

**Comment by PiotrSikora** [comment]
(`main` branch is broken, that's why the format check fails...)

**Comment by desimone** [comment]
LMK if you need anything additional on my end

**Comment by htuch** [comment]
Try mergre main and push again. Thanks.

**Comment by desimone** [comment]
`git merge origin/main && git push` ✅

**Review by htuch — COMMENTED** [review]
LGTM, thanks!

**Pull request merged**
